core/src/test/java/com/graphhopper/routing/util/spatialrule/SpatialRuleLookupArrayTest.java

```diff
@@ -0,0 +1,203 @@
+        Polygon germanPolygon = parsePolygonString(germanPolygonJson);
+
+        SpatialRule germanySpatialRule = new GermanySpatialRule();
+        germanySpatialRule.addBorder(germanPolygon);
+        spatialRuleLookup.addRule(germanySpatialRule);
+
+        // Taken from here: https://github.com/johan/world.geo.json/blob/master/countries/AUT.geo.json
+        String austrianPolygonJson = "[16.979667,48.123497],[16.903754,47.714866],[16.340584,47.712902],[16.534268,47.496171],[16.202298,46.852386],[16.011664,46.683611],[15.137092,46.658703],[14.632472,46.431817],[13.806475,46.509306],[12.376485,46.767559],[12.153088,47.115393],[11.164828,46.941579],[11.048556,46.751359],[10.442701,46.893546],[9.932448,46.920728],[9.47997,47.10281],[9.632932,47.347601],[9.594226,47.525058],[9.896068,47.580197],[10.402084,47.302488],[10.544504,47.566399],[11.426414,47.523766],[12.141357,47.703083],[12.62076,47.672388],[12.932627,47.467646],[13.025851,47.637584],[12.884103,48.289146],[13.243357,48.416115],[13.595946,48.877172],[14.338898,48.555305],[14.901447,48.964402],[15.253416,49.039074],[16.029647,48.733899],[16.499283,48.785808],[16.960288,48.596982],[16.879983,48.470013],[16.979667,48.123497]";
+        Polygon austrianPolygon = parsePolygonString(austrianPolygonJson);
+
+        SpatialRule austrianSpatialRule = new AustriaSpatialRule();
+        austrianSpatialRule.addBorder(austrianPolygon);
+        spatialRuleLookup.addRule(austrianSpatialRule);
+
+
+
+        ReaderWay track = new ReaderWay(0);
+        track.setTag("highway", "track");
+
+        ReaderWay primary = new ReaderWay(0);
+        primary.setTag("highway", "primary");
+
+        ReaderWay livingStreet = new ReaderWay(0);
+        livingStreet.setTag("highway", "living_street");
+
+        // Berlin
+        assertEquals(AccessValue.EVENTUALLY_ACCESSIBLE, spatialRuleLookup.lookupRule(52.5243700, 13.4105300).isAccessible(track, ""));
+        assertEquals(AccessValue.ACCESSIBLE, spatialRuleLookup.lookupRule(52.5243700, 13.4105300).isAccessible(primary, ""));
+
+        // Paris
+        assertEquals(AccessValue.ACCESSIBLE, spatialRuleLookup.lookupRule(48.864716, 2.349014).isAccessible(track, ""));
+        assertEquals(AccessValue.ACCESSIBLE, spatialRuleLookup.lookupRule(48.864716, 2.349014).isAccessible(primary, ""));
+
+        // Vienna
+        assertEquals(AccessValue.ACCESSIBLE, spatialRuleLookup.lookupRule(48.210033, 16.363449).isAccessible(track, ""));
+        assertEquals(AccessValue.ACCESSIBLE, spatialRuleLookup.lookupRule(48.210033, 16.363449).isAccessible(primary, ""));
+        assertEquals(AccessValue.EVENTUALLY_ACCESSIBLE, spatialRuleLookup.lookupRule(48.210033, 16.363449).isAccessible(livingStreet, ""));
+    }
+
+    private Polygon parsePolygonString(String polygonString){
```

Comment:
Do you think this would be the better approach? I tried to keep the core clean of JSON.